docs/config/simple.md

```diff
@@ -1,208 +1,344 @@
 
 ```text
-pattern: foo("...")
-
-# CODE EXAMPLES
+pattern: crypto.set_secret_key("...")
+```
 
-foo("this is a specific string")
+```python
+crypto.set_secret_key("HARDCODED SECRET")
 ```
 
-**Using** [**OCaml regular expression**](https://caml.inria.fr/pub/docs/manual-ocaml/libref/Str.html) **Patterns**
+[OCaml regular expressions](https://caml.inria.fr/pub/docs/manual-ocaml/libref/Str.html)
+can also be used to search for specific strings:
 
 ```text
-pattern: foo("=~/.*a.*/")
+pattern: requests.get("=~/.*dev\.corp\.com.*/")
+```
+
+```python
+requests.get("api.dev.corp.com")  # Oops, development API left in
+```
+
+#### Arrays
 
-# CODE EXAMPLES
+The ellipsis operator can be used to match literal arrays:
 
-foo("this has an a")
+```text
+pattern: user_list = [..., 10]
+```
+
+```python
+user_list = [8, 9, 10]
 ```
 
-### Conditionals
+#### Conditionals and Loops
+
+The ellipsis operator can be used inside conditionals or loops:
 
 ```text
 pattern: |
-    if $X:
-        $Y
+  if $CONDITION:
+      ...
+```
 
-# CODE EXAMPLES
+*Note the YAML `|` operator allows for [multiline strings](https://yaml-multiline.info/).*
 
-if __name__ == "__main__":
-    print('hello world')
+```python
+if can_make_request:
+    check_status()
+    make_request()
+    return
 ```
 
+A metavariable can also be used to match a conditional or loop body if the
+body statement information will be re-used later:
+
 ```text
 pattern: |
-    if $X:
-        ...
-
-# CODE EXAMPLES
+  if $CONDITION:
+      $BODY
+```
 
-if __name__ == "__main__":
-    print('hello world')
-    foo()
-    bar()
+```python
+if can_make_request:
+    single_request_statement()
 ```
 
-Note you can’t match a half statement; both of the examples above must specify the contents of the condition’s body \(e.g. `$Y` and `...`\), otherwise they are not valid AST elements.
+**Note you can’t match a half statement; both of the examples above must
+specify the contents of the condition's body (e.g. `$BODY` or `...`),
+otherwise they are not valid patterns.**
+
+### Equivalences
+
+`semgrep` will handle certain types of equivalent code.
 
-### In a statement context, a Metavariable can also match any statement
+#### Imports
+
+Equivalent imports using aliasing or submodules will be matched.
 
 ```text
-pattern: |
-    if $X:
-        $Y
+pattern: subprocess.Popen(...)
+```
 
-# CODE EXAMPLES
-if 1:
-  foo()
+```python
+import subprocess.Popen as sub_popen
+sub_popen('ls')
+```
 
-if 2:
-  return 1
+```text
+pattern: foo.bar.baz.qux(...)
+```
 
-if 3:     # matches a “block” (a single statement containing multiple statements)
-  foo()
-  bar()
+```python
+from foo.bar import baz
+baz.qux()
 ```
 
-Because in Python there is usually no terminator \(e.g., `;`\), there is an ambiguity about `$Y` in the above, which could match a statement and also an expression that is then matched later.
+#### Constants
 
-### Match on import types
+JavaScript code will match propagated constants:
```

Comment:
Golang works too. Maybe say "some languages" to make it more generic? not all languages have constants

Reply:
Good call, I'll add a Golang example too